virtcontainers/sandbox.go

```diff
@@ -1201,6 +1207,42 @@ func (s *Sandbox) CreateContainer(contConfig ContainerConfig) (VCContainer, erro
 	return c, nil
 }
 
+func (s *Sandbox) creditContainerVCPU(num uint32, credit bool) (uint32, error) {
```

Comment:
I think this function could do with an explanatory comment (which should explain what `credit` is and what the returned int value represents).

I also find the name a little confusing since:

- it's not working on a particular container - it's working at the pod level.
- the name doesn't really fully explain what it's doing. Maybe something like `adjustVCPUCount()`?

Lastly, please can you add a unit-test for the calculation code (maybe by putting that into a new function to make it easier to test).